### DISCUSSION TITLE
Using ES6 import in Node

### DISCUSSION THREAD

**Discussion by vit1251** [slint-ui/slint#1406] [Q&A]
Let's try to start `slint-ui` and break an at start with using outdated `require` keyword.
Do u support ES6 specification like:

```
import slint from "slint-ui";
import ui from "ui/main.slint";

let component = new ui.MainWindow({
    counter: 42,
    clicked: () => {
        console.log("hello");
    },
});
```

How to enable ES6? Any example exists?

**Comment by tronical** [comment]
That’s a really good question. We use `require.extensions` at the moment to hook into the load mechanism of nodejs. I wonder how to hook into the esm import mechanism.

I see dynamic transpiling as example in the docs, but it requires a command line option: https://nodejs.org/dist/latest-v18.x/docs/api/esm.html#transpiler-loader

Does anybody know?:)

**Reply by vit1251** [reply]
Since we does not have a legal ES6 loading and use mechanism may I ask you help me with create bundle manual by invoke some specific `slint` function in `npm` scripts or even on fly? I use before JSX subsystem on run with `babel` function `transformSync` may be `slint` also have a similar function?

**Reply by tronical** [reply]
This on the fly transform is what [this](https://github.com/slint-ui/slint/blob/ced950463542e138abf362c4d206897bc9e29783/api/node/lib/index.ts#L113) kind of does. Take a Slint file and transform it into a JS object.

**Reply by vit1251** [reply]
Let's try to play with this code. Thanks.
